### PR TITLE
[release-v1.12] Update vendor patch list

### PR DESCRIPTION
**Pull request by matzew**
A few vendor patches are missing on the `release-v1.12` branch, so that `make generate-release` would not re-apply them

* For #1056 the patch for the backport of a `vendor` fix was not added as patch, just `vendor` patch

### CODE REVIEW

File: openshift/release/generate-release.sh [hunk 1]
```diff
@@ -5,6 +5,8 @@ set -euo pipefail
 source $(dirname $0)/resolve.sh
 
 GITHUB_ACTIONS=true $(dirname $0)/../../hack/update-codegen.sh
+git apply openshift/patches/disable-ko-publish-rekt.patch
```
Comment (pierDipi) [inline openshift/release/generate-release.sh:8]: This is ok to not be applied, we can remove it

Comment (pierDipi) [reply openshift/release/generate-release.sh:8]: We're using the images file flag and so ko will not be used

Comment (pierDipi) [reply openshift/release/generate-release.sh:8]: ref for main https://github.com/openshift-knative/eventing/pull/605

Comment (matzew) [reply openshift/release/generate-release.sh:8]: Ok. thanks - I guess it should be consistent than. I will update.

**Pull request merged**